ad9959/src/lib.rs

```diff
@@ -216,23 +216,16 @@ impl<I: Interface> Ad9959<I> {
         reference_clock_frequency: f32,
         multiplier: u8,
     ) -> Result<f32, Error> {
+        let frequency =
+            validate_clocking(reference_clock_frequency, multiplier)?;
         self.reference_clock_frequency = reference_clock_frequency;
 
-        if multiplier != 1 && !(4..=20).contains(&multiplier) {
-            return Err(Error::Bounds);
-        }
-
-        let frequency = multiplier as f32 * self.reference_clock_frequency;
-        if frequency > 500_000_000.0f32 {
-            return Err(Error::Frequency);
-        }
-
         // TODO: Update / disable any enabled channels?
         let mut fr1: [u8; 3] = [0, 0, 0];
         self.read(Register::FR1, &mut fr1)?;
         fr1[0].set_bits(2..=6, multiplier);
 
-        let vco_range = frequency > 255e6;
+        let vco_range = frequency > 200e6;
```

Comment:
Please move the 200e6 to a `const` variable with a name at the top of the file. Also, can you please document where this limit is coming from? It's not clear to me (nor can I remember why the original 255e6 was selected)

Reply:
Documented. Its selection is somewhat arbitrary.

Reply:
I talk about this below, but we should just use the min of the high VCO gain config or max of the low VCO gain config constants instead of this arbitrary value.